src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/cdk/cdk_stacks/adf_standard_pipelines.py

```diff
@@ -0,0 +1,260 @@
+            )
+        elif "s3" in target_stage_override:
+            _actions.extend(
+                [
+                    adf_codepipeline.Action(
+                        name="{0}-{1}".format(target["name"], region),
+                        provider="S3",
+                        category="Deploy",
+                        region=region,
+                        target=target,
+                        action_mode=top_level_action,
+                        run_order=1,
+                        map_params=stack_input["input"],
+                        action_name="{0}-{1}".format(target["name"], region),
+                    ).config
+                ]
+            )
+        elif "lambda" in target_stage_override:
+            _actions.extend(
+                [
+                    adf_codepipeline.Action(
+                        name="{0}-{1}".format(target["name"], region),
+                        provider="Lambda",
+                        category="Invoke",
+                        region=region,
+                        target=target,
+                        action_mode=top_level_action,
+                        run_order=1,
+                        map_params=stack_input["input"],
+                        action_name="{0}-{1}".format(target["name"], region),
+                    ).config
+                ]
+            )
+        elif "service_catalog" in target_stage_override:
+            _actions.extend(
+                [
+                    adf_codepipeline.Action(
+                        name="{0}-{1}".format(target["name"], region),
+                        provider="ServiceCatalog",
+                        category="Deploy",
+                        region=region,
+                        target=target,
+                        action_mode=top_level_action,
+                        run_order=1,
+                        map_params=stack_input["input"],
+                        action_name="{0}-{1}".format(target["name"], region),
+                    ).config
+                ]
+            )
+
+
+def generate_build_stage_for_pipeline(_stages, scope, stack_input):
+    _build_name = (
+        stack_input["input"]["default_providers"]["build"].get("provider", "").lower()
+    )
+    if "codebuild" in _build_name and stack_input["input"]["default_providers"][
+            "build"
+    ].get("enabled", True):
+        _stages.append(
+            adf_codebuild.CodeBuild(
+                scope,
+                "build",
+                stack_input["ssm_params"][ADF_DEPLOYMENT_REGION]["modules"],
+                stack_input["ssm_params"][ADF_DEPLOYMENT_REGION]["kms"],
+                stack_input["input"],
+                {},  # Empty target since this is a build only stage
+            ).build
+        )
+    elif "jenkins" in _build_name:
+        _stages.append(adf_jenkins.Jenkins(scope, "build", stack_input["input"]).build)
+
+
+def generate_source_stage_for_pipeline(_stages, scope, stack_input):
```

Comment:
Could you move the functions in the order as they will be executed, so source first, build second, targets third, actions fourth, etc?